src/Compatibility/ApiDiff/Microsoft.DotNet.ApiDiff.Tool/Program.cs

```diff
@@ -0,0 +1,172 @@
+                                              optionAfterRefAssembliesFolderPath,
+                                              optionAttributesToExclude,
+                                              optionBeforeAssembliesFolderPath,
+                                              optionBeforeRefAssembliesFolderPath,
+                                              optionDebug,
+                                              optionHideImplicitDefaultConstructors,
+                                              optionIncludeTableOfContents,
+                                              optionOutputFolderPath);
+
+        rootCommand.SetHandler(HandleCommand, c);
+        await rootCommand.InvokeAsync(args);
+    }
+
+    private static void HandleCommand(DiffConfiguration diffConfig)
+    {
+        var log = new ConsoleLog(MessageImportance.Normal);
+
+        // Custom ordering to match help menu.
+        log.LogMessage("Selected options:");
+        log.LogMessage($" - 'Before' assemblies:                {diffConfig.BeforeAssembliesFolderPath}");
+        log.LogMessage($" - 'Before' reference assemblies:      {diffConfig.BeforeAssemblyReferencesFolderPath}");
+        log.LogMessage($" - 'After' assemblies:                 {diffConfig.AfterAssembliesFolderPath}");
+        log.LogMessage($" - 'After' ref assemblies:             {diffConfig.AfterAssemblyReferencesFolderPath}");
+        log.LogMessage($" - Output:                             {diffConfig.OutputFolderPath}");
+        log.LogMessage($" - Attributes to exclude:              {string.Join(", ", diffConfig.AttributesToExclude)}");
+        log.LogMessage($" - Include table of contents:          {diffConfig.IncludeTableOfContents}");
+        log.LogMessage($" - Add partial modifier to types:      {diffConfig.AddPartialModifier}");
+        log.LogMessage($" - Hide implicit default constructors: {diffConfig.HideImplicitDefaultConstructors}");
+        log.LogMessage($" - Debug:                              {diffConfig.Debug}");
+        log.LogMessage("");
+
+        if (diffConfig.Debug)
+        {
+            WaitForDebugger();
+        }
+
+        Dictionary<string, string> results = DiffGenerator.Run(log,
+            diffConfig.AttributesToExclude,
+            diffConfig.BeforeAssembliesFolderPath,
+            diffConfig.BeforeAssemblyReferencesFolderPath,
+            diffConfig.AfterAssembliesFolderPath,
+            diffConfig.AfterAssemblyReferencesFolderPath,
+            diffConfig.AddPartialModifier,
+            diffConfig.HideImplicitDefaultConstructors);
+
+        Directory.CreateDirectory(diffConfig.OutputFolderPath);
+        foreach ((string assemblyName, string text) in results)
+        {
+            string filePath = Path.Combine(diffConfig.OutputFolderPath, $"{assemblyName}.md");
+            File.WriteAllText(filePath, text);
+            log.LogMessage($"Wrote '{filePath}'.");
+        }
```

Comment:
Similar comment as above. I think it would make sense to move this into the business layer to keep the UI layer as thin as possible.

Reply:
Mmm let me see what would be the best way to do this, as the intention was to be able to avoid writing to disk in the unit tests.

Reply:
I did it by just adding another public method that runs the tool by writing to disk.

Reply:
I'm exploring using a strategy as suggested.

Reply:
Ok, I implemented it.